Update esp8266-weather-station-color.ino

TiVoHomeUser opened the pull request:
- Allow user to force a screen re-calibration
 - Break up the screen into 4 sections for navigation

Fixes #148.

- [x] This PR is compliant with the [other contributing guidelines](https://github.com/ThingPulse/esp8266-weather-station-color/blob/master/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] Should this code require changes to documentation I will contribute those to [https://github.com/ThingPulse/docs](https://github.com/ThingPulse/docs).

\<Description of and rationale behind this PR\>

marcelstoer commented:
I just tested this myself and I still like your idea. Thanks!

For whatever reason my test device here has got top/bottom mixed up for the touch screen. Don't understand this yet as the whole point of the calibration procedure is to take care of this. Sigh...

I'll leave review comments right with the code.

marcelstoer commented on esp8266-weather-station-color.ino:
```diff
@@ -238,15 +260,38 @@ uint8_t currentTouchPoint = 0;
   gfx.fillBuffer(MINI_BLACK);
-  if (touchController.isTouched(0)) {
+
+  /* Break up the screen into 4 sections a touch in section:
```
I think it would finally be time to document the touch behavior in the README 😉 Even the 12/24h switching was documented - shame on us.

TiVoHomeUser replied:
I updated the README for the force calibration, touch screen points and added an image

marcelstoer commented on esp8266-weather-station-color.ino:
```diff
@@ -238,15 +260,38 @@ uint8_t currentTouchPoint = 0;
+   * - Right forward one page
+   * - Bottom jump to page 0
+   */
+  const static uint8_t top = gfx.getHeight() / 4;
```
Kind-of unrelated to this PR but...

Is it worth the effort fetching width & height dynamically? `TouchControllerWS.cpp` has 240/320 embedded in the code 🙈 Maybe extract that into `settings.h`, WDYT?

TiVoHomeUser replied:
I don't think that hard coding the screen size is good. The screen is defiantly something that should be runtime calibrated if it is a fixed size why would touch calibration be needed?

marcelstoer commented on esp8266-weather-station-color.ino:
```diff
@@ -238,15 +260,38 @@ uint8_t currentTouchPoint = 0;
-    if (!asleep) { // no need to change screens;
-      if (p.y < 80) {
+    if (!asleep) { // no need to update or change screens;
+      if (p.y <= top)   Serial.print(" TOP ");
```
Since these are not if-else-if you might see "Bottom Right" on the console even though no action is associated with that specific section (only "Bottom" is relevant in that context). That made me wonder...

Ideally 
- all of that logic should go into a "ScreenSection"-like class or function
- the evaluation of which section was clicked should accept a `TS_Point` and return a section enum: TOP, BOTTOM, MIDDLE_LEFT, MIDDLE_RIGHT

TiVoHomeUser replied:
- That is by design for the debugging output the extra space in the string gives output like " Top Left " or " Right Bottom " the screen selection part is using the nested if then else.
- Agree it would be better to do the screen selection and draw screen in a function call outside of loop().
- using enum would allow using case instead of if/then/else.
I'll work on making these changes, thanks.

TiVoHomeUser replied:
Any thing else I need to do?

TiVoHomeUser commented:
Moved touch screen point selection to new changeScreen function.
Commented out confusing screen selection debug output.
changed unsigned 16 bit member variable screen to unsigned 8 bit.

TiVoHomeUser commented:
**marcelstoer:**  I see the size I used was for the display object not the touch screen your changes with the fixed size looks good.

marcelstoer commented:
Thanks again for the inspiration and the first implementation!
